internalshared/configutil/normalize.go

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package configutil
+
+import (
+	"fmt"
+	"net"
+	"net/url"
+	"strings"
+)
+
+// NormalizeAddr takes an address as a string and returns a normalized copy.
+// If the addr is a URL, IP Address, or host:port address that includes an IPv6
+// address, the normalized copy will be conformant with RFC-5942 §4
+// See: https://rfc-editor.org/rfc/rfc5952.html
+func NormalizeAddr(u string) string {
```

Comment:
I'd like if we chose different variable name than `u` here. Maybe `addressToNormalize`?

Reply:
I decided to go with  `address` here as it was certainly better than `u` (a relic of the genesis of the func when it only took URLs) but less typey than `addressToNormalize`.